src/util/normalizeThought.ts

```diff
@@ -1,31 +1,12 @@
-import emojiStrip from 'emoji-strip'
-import * as pluralize from 'pluralize'
 import _ from 'lodash'
-import { REGEXP_TAGS } from '../constants'
+import { singularize } from './singularize'
+import { stripTags } from './stripTags'
+import { trim } from './trim'
+import { stripEmojiFromText } from './stripEmojiFromText'
 
 /** Converts a string to lowecase. */
 const lower = (s: string) => s.toLowerCase()
 
-/** Trims a string. */
-const trim = (s: string) => s.replace(
-  s.length > 0 && s.replace(/\W/g, '').length > 0 ? /\W/g : /s/g,
-  ''
-)
-
-/** Strips emoji from text. Preserves emoji on its own. */
-const stripEmojiWithText = (s: string) => {
-  const stripped = emojiStrip(s)
-  return stripped.length > 0 ? stripped : s
-}
-
-/** Strips all html tags. */
-const stripTags = (s: string) => s.replace(REGEXP_TAGS, '')
-
-/**
- * Making character 's' will just become an empty value ''.
- * Skip it else it will cause "s" character to have same no of context as empty thoughts in the entire tree. */
-const singularize = (s: string) => s !== 's' ? pluralize.singular(s) : s
-
```

Comment:
I would prefer to keep these functions in this module. It keeps the definitions close to their use and minimizes the surface area of the project's internal API. Then if you need to use one of these functions in a separate module then you can factor it out.

Reply:
Understood.